src/web/routes/application/select-address/adapters.js

```diff
@@ -0,0 +1,76 @@
+const { props, propOr, pickAll, compose, map, join, filter, not, isEmpty, split, head, tail, defaultTo } = require('ramda')
+const { OS_PLACES_ADDRESS_KEYS } = require('../common/constants')
```

Comment:
One for another PR perhaps, but could OS_PLACES_ADDRESS_KEYS have constants for each key, and another for all keys? Then we could do:
```
const constructBuildingName = joinAddressFields([OS_PLACES_ADDRESS_KEYS.SUB_BUILDING_NAME, OS_PLACES_ADDRESS_KEYS.BUILDING_NAME])
```
and
```
const normaliseAddressFields = compose(map(defaultTo('')), pickAll(OS_PLACES_ADDRESS_KEYS.ALL))
```

Reply:
I realise OS_PLACES_ADDRESS_KEYS.SUB_BUILDING_NAME is a bit wordy

Reply:
I'm not sure I totally understand but sounds like it's a good idea. It feels like this has implications for the postcode page as well so I will merge this PR and then look at creating another PR to address this (I think it also relates to your comment below).